### PR TITLE
Avoid using `_typeshed.StrEnum`

### PR DESCRIPTION
**Pull request by tungol**
This was a perfectly reasonable way to handle StrEnum. That said, there's only two uses of it, and this is just about the only thing currently in the stdlib that benefits from the complicated MRO-with-skips comparison in the original version of @hauntsaninja's inheritance checker from https://github.com/python/typeshed/issues/3968. As I get ready to propose a version of that check for inclusion in stubtest, it'd be nice to drop that part.

### CODE REVIEW

File: stdlib/_typeshed/__init__.pyi [hunk 1]
```diff
@@ -352,12 +352,3 @@ class DataclassInstance(Protocol):
 # Anything that can be passed to the int/float constructors
 ConvertibleToInt: TypeAlias = str | ReadableBuffer | SupportsInt | SupportsIndex | SupportsTrunc
 ConvertibleToFloat: TypeAlias = str | ReadableBuffer | SupportsFloat | SupportsIndex
-
-# A few classes updated from Foo(str, Enum) to Foo(StrEnum). This is a convenience so these
-# can be accurate on all python versions without getting too wordy
-if sys.version_info >= (3, 11):
-    from enum import StrEnum as StrEnum
```
Comment (sobolevn) [inline stdlib/_typeshed/__init__.pyi:359]: This might be used by others as well. Why do we need to break `_typeshed.StrEnum` usage for custom stubs?

Comment (tungol) [reply stdlib/_typeshed/__init__.pyi]: I put it back in `_typeshed`; I hadn't thought there would be users outside of typeshed itself

Comment (srittau) [reply stdlib/_typeshed/__init__.pyi]: But please note that it's not marked as stable, so we *can* break it if we intend to do so. Personally, I didn't mind either way, although as far as I understand it, this is mainly a workaround around a mypy bug, so we might remove it in the future once the bug is resolved.

Comment (tungol) [reply stdlib/_typeshed/__init__.pyi]: I don't think there's mypy bug involved here. `_typeshed.StrEnum` is just for convenience purposes in writing stubs that use `enum.StrEnum`, and can certainly go away once all supported python versions support that.

**Review by srittau — APPROVED** [review]
